cabal-install/Distribution/Client/JobControl.hs

```diff
@@ -27,48 +27,129 @@ module Distribution.Client.JobControl (
   ) where
 
 import Control.Monad
-import Control.Concurrent hiding (QSem, newQSem, waitQSem, signalQSem)
-import Control.Exception (SomeException, bracket_, mask, throw, try)
+import Control.Concurrent (forkIO) --hiding (QSem, newQSem, waitQSem, signalQSem)
+import Control.Concurrent.MVar
+import Control.Concurrent.STM (STM, atomically)
+import Control.Concurrent.STM.TVar
+import Control.Concurrent.STM.TQueue
```

Comment:
Note this uses `TQueue` from `stm-2.4`. Are we ok with that version? Alternative could probably use `TChan`.